mainchain-follower/db-sync-follower/src/db_model.rs

```diff
@@ -404,9 +404,11 @@ pub(crate) async fn get_latest_block_for_epoch(
 	pool: &Pool<Postgres>,
 	epoch: EpochNumber,
 ) -> Result<Option<Block>, SqlxError> {
+	// Query below contains additional filters for slot_no and block_no not null, because
+	// there exists blocks in Byron Era with Ouroboros classic consensus that have null values for these fields.
 	let sql = "SELECT block.block_no, block.hash, block.epoch_no, block.slot_no, block.time
 		FROM block
-		WHERE block.epoch_no <= $1
+		WHERE block.epoch_no <= $1 AND block.slot_no IS NOT NULL AND block.block_no IS NOT NULL
```

Comment:
I was thinking about adding additional test for proto_major > 0 but db-sync implementation has a history of using strange combinations of proto_major and proto_minor for some internal stuff. So, I have decided to filter these blocks by slot or block number being null.